### PR TITLE
Refactor okd

### PR DESCRIPTION
**Pull request by cynepco3hahue**
Refactor OKD provision and run scripts
    
- move registry yaml configuration to the manifests directory
- move install-config.yaml from base container to manifests directory
- remove pull-token after provision
- ask for pull-token on run time
- remove if condition, what works for 4.3 also works for 4.1
- add the `--installer-pull-secret-file` parameter to the run CLI
- change the gocli package prefix

### CODE REVIEW

File: cluster-provision/okd/base/Dockerfile [hunk 1]
```diff
@@ -1,24 +1,36 @@
...
-httpd-tools && \
+httpd-tools \
+python2-pip && \
 dnf clean all
 
+RUN pip2 install yq
+
+# install golang
+ENV GO_VERSION 1.12.12
+RUN curl https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz -o go.tar.gz && \
```
Comment (qinqon) [inline cluster-provision/okd/base/Dockerfile]: This is going to make okd- providers bigger, but not much though, is that ok ?

**Comment by oshoval** [comment]
Quique and i experienced a problem with /root/.ssh not found 
usually after something else major happend and the script had to collect logs
checking the container it indeed didnt have files at that folder

so maybe worth to add a tuple, either the exiting vagrant one (with renaming it)
or just to add 
ssh-keygen -t rsa -f /root/.ssh/id_rsa -N ''
before cluster create ?

thanks

File: cluster-provision/okd/base/Dockerfile (hunk 1 above)
Comment (cynepco3hahue) [reply cluster-provision/okd/base/Dockerfile]: the difference is 300MB, IMHO it is not critical when we have images of 14Gb :(

**Comment by cynepco3hahue** [comment]
> Quique and i experienced a problem with /root/.ssh not found
> usually after something else major happend and the script had to collect logs
> checking the container it indeed didnt have files at that folder
> 
> so maybe worth to add a tuple, either the exiting vagrant one (with renaming it)
> or just to add
> ssh-keygen -t rsa -f /root/.ssh/id_rsa -N ''
> before cluster create ?
> 
> thanks

@oshoval Can you please open an issue and I will fix it under the next iteration?

**Comment by oshoval** [comment]
> > Quique and i experienced a problem with /root/.ssh not found
> > usually after something else major happend and the script had to collect logs
> > checking the container it indeed didnt have files at that folder
> > so maybe worth to add a tuple, either the exiting vagrant one (with renaming it)
> > or just to add
> > ssh-keygen -t rsa -f /root/.ssh/id_rsa -N ''
> > before cluster create ?
> > thanks
> 
> @oshoval Can you please open an issue and I will fix it under the next iteration?

sure thanks
https://github.com/kubevirt/kubevirtci/issues/188

**Comment by qinqon** [comment]
@cynepco3hahue looks like openshift installer libvirt is broken https://github.com/openshift/installer/pull/2657

**Pull request merged**